src/sql/src/query_model/hir.rs

```diff
@@ -431,3 +452,137 @@ impl FromHir {
         }
     }
 }
+
+/// Special transformations to fix semantically invalid constructions allowed
+/// temporarily for the sake of keeping the generator above as simple as possible.
+mod fixup {
```

Comment:
I'd say 3. This is not a query rewrite per se, since it's only needed to prevent the generator from returning something semantically invalid. We could go with 1 as well, if we find a way of integrating this logic everytime we generate a Select box.